command/job_deployments.go

```diff
@@ -122,10 +122,17 @@ func (c *JobDeploymentsCommand) Run(args []string) int {
 		c.Ui.Error(fmt.Sprintf("No job(s) with prefix or id %q found", jobID))
 		return 1
 	}
-	if len(jobs) > 1 && (c.allNamespaces() || strings.TrimSpace(jobID) != jobs[0].ID) {
-		c.Ui.Error(fmt.Sprintf("Prefix matched multiple jobs\n\n%s", createStatusListOutput(jobs, c.allNamespaces())))
-		return 1
+	if len(jobs) > 1 {
+		if strings.TrimSpace(jobID) != jobs[0].ID {
```

Comment:
nit: couldn't we move the TrimSpace up above, i.e. `jobID := strings.TrimSpace(args[0])`

Reply:
Alternately, we could remove it and punish anyone who intentionally does `nomad job status " example "` 😀

Will fix!